tests/ManageCourses.Tests/DbIntegration/DataServiceTests.cs

```diff
@@ -250,5 +253,264 @@ public void GetCoursesForUser_isNull(string email, string orgId)
+            LoadData(TestUserEmail_1, numOrgs, numCourses);
+            var result = Subject.GetOrganisationsForUser(TestUserEmail_1).ToList();
+            Assert.IsTrue(result.Count == numOrgs);
+            Assert.IsTrue(result.All(c => c.TotalCourses == numCourses));
+        }
+        [Test]
+        public void GetOrganisationsShouldReturnNoData()
+        {
+            const int numOrgs = 5;
+            const int numCourses = 6;
+            LoadData(TestUserEmail_1, numOrgs, numCourses);
+            var result = Subject.GetOrganisationsForUser("anyone@testing.com").ToList();//try to get the list using an invalid email
+            Assert.IsTrue(result.Count == 0);
+            Assert.IsTrue(result.All(c => c.TotalCourses == 0));
+        }
+        [Test]
+        public void GetOrganisationShouldReturnData()
+        {
+            const int numOrgs = 5;
+            const int numCourses = 6;
+            LoadData(TestUserEmail_1, numOrgs, numCourses);
+            var orgList = Subject.GetOrganisationsForUser(TestUserEmail_1).ToList();
+
+            Assert.IsTrue(orgList.Count == numOrgs);
+            Assert.IsTrue(orgList.All(c => c.TotalCourses == numCourses));
+
+            foreach (var org in orgList)//we have a valid list of data
+            {
+                var result = Subject.GetOrganisationForUser(TestUserEmail_1, org.UcasCode);//get the organisation
+                Assert.IsTrue(result.OrganisationName == org.OrganisationName);
+            }
+        }
+        [Test]
+        public void GetOrganisationShouldReturnNull()
+        {
+            const int numOrgs = 5;
+            const int numCourses = 6;
+            LoadData(TestUserEmail_1, numOrgs, numCourses);
+            var orgList = Subject.GetOrganisationsForUser(TestUserEmail_1).ToList();
+            Assert.IsTrue(orgList.Count == numOrgs);
+            Assert.IsTrue(orgList.All(c => c.TotalCourses == numCourses));
+
+            foreach (var org in orgList)//we have a valist list of data
+            {
+                var result = Subject.GetOrganisationForUser("anyone@testing.com", org.UcasCode);//try to get the organisation using an invalid email
+                Assert.IsNull(result);
+            }
+        }
+        [Test]
+        public void GetCoursesShouldReturnData()
+        {
+            const int numOrgs = 5;
+            const int numCourses = 6;
+            LoadData(TestUserEmail_1, numOrgs, numCourses);
+            var orgList = Subject.GetOrganisationsForUser(TestUserEmail_1).ToList();
+            Assert.IsTrue(orgList.Count == numOrgs);
+            Assert.IsTrue(orgList.All(c => c.TotalCourses == numCourses));
+
+            foreach (var org in orgList)//we have a valist list of data
+            {
+                var result = Subject.GetCourses(TestUserEmail_1, org.UcasCode);//get the course for each org
+                Assert.IsTrue(result != null && result.Courses.Count == numCourses);
```

Comment:
Can we start breaking it out as 
`Assert.IsNotNull(result)` <= that can be obmitted as the below line actual needs it
`Assert.AreEqual(numCourses , result.Courses.Count)`

Reply:
I agree. Much cleaner